spec/models/user_notification_spec.rb

```diff
@@ -0,0 +1,44 @@
+# == Schema Information
+#
+# Table name: user_notification
+#
+#  course_update :boolean          default("false")
+#  created_at    :datetime         not null
+#  id            :bigint           not null, primary key
+#  provider_code :string           not null
+#  updated_at    :datetime         not null
+#  user_id       :integer          not null
+#
+describe UserNotification, type: :model do
+  describe "associations" do
+    let(:organisation) { create(:organisation, providers: [provider]) }
+    let(:user) { create(:user, organisations: [organisation]) }
+    let(:provider) { create(:provider) }
+
+    subject { UserNotification.new(user_id: user.id, provider_code: provider.provider_code) }
```

Comment:
```suggestion
    subject { described_class.new(user_id: user.id, provider_code: provider.provider_code) }
```

maybe use `described_class`